controllers/provider-gcp/pkg/internal/infrastructure/terraform.go

```diff
@@ -177,6 +204,9 @@ func StatusFromTerraformState(state *TerraformState) *gcpv1alpha1.Infrastructure
 			Networks: gcpv1alpha1.NetworkStatus{
 				VPC: gcpv1alpha1.VPC{
 					Name: state.VPCName,
+					CloudRouter: &gcpv1alpha1.CloudRouter{
```

Comment:
Only write the cloud router name into the status if it was created or provided by the end-user